### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by shmool**

### CODE REVIEW

File: src/app/rappid-components/rappid-toolbar/rappid-toolbar.component.ts [hunk 1]
```diff
@@ -1,31 +1,72 @@
 import { Component, OnInit } from '@angular/core';
 import { GraphService } from '../services/graph.service';
-import { SaveModelDialogComponent } from '../../dialogs/save-model-dialog/save-model-dialog.component';
 import { MdDialog } from '@angular/material';
 import { LoadModelDialogComponent } from '../../dialogs/load-model-dialog/load-model-dialog.component';
 import { CommandManagerService } from '../services/command-manager.service';
-import {ModelObject} from "../../services/storage/model-object.class";
+
+const commandGroups = [
```
Comment (NataliLevi) [inline src/app/rappid-components/rappid-toolbar/rappid-toolbar.component.ts]: Need to write a function that build and return a block:
 {
 +    group: 'editor',
 +    commands: [
 +      { name: 'undo', tooltip: 'undo', icon: 'undo' },
 +      { name: 'redo', tooltip: 'redo', icon: 'redo' }
 +    ]
 +  }
and caal it each time with different parameters.

File: src/app/rappid-components/services/init-rappid.service.ts [hunk 2]
```diff
@@ -427,53 +382,57 @@ export class RappidMainComponent implements OnInit {
...
     this.graph.on('change:attrs', function (cell) {
-      if((cell.get('type') != 'opm.Link') && (cell.attr('text/text') != lastEnteredText) &&
-          !cell.attributes.attrs.wrappingResized){  //if the text was changed
+      if ((cell.get('type') != 'opm.Link') && (cell.attr('text/text') != lastEnteredText) &&
+        !cell.attributes.attrs.wrappingResized) {  //if the text was changed
         var textString = cell.attr('text/text');
-        var newParams = {width: cell.get('minSize').width, height: cell.get('minSize').height, text: '\t'};    //No empty name is allowed
-        if(textString.trim() != '') { //if there is areal text - not spaces
+        var newParams = { width: cell.get('minSize').width, height: cell.get('minSize').height, text: '\t' };    //No
+                                                                                                                 // empty
```
Comment (NataliLevi) [inline src/app/rappid-components/services/init-rappid.service.ts]: empy text is not allowed - Dov's requirement

File: src/environments/environment.ts [hunk 3]
```diff
@@ -4,5 +4,16 @@
 // The list of which env maps to which file can be found in `angular-cli.json`.
 
 export const environment = {
-  production: false
+  production: false,
+  firebaseCredentials: {
```
Comment (NataliLevi) [inline src/environments/environment.ts]: why firebaseCredentials defined twice? in this file and in firbase.config.ts?

Comment (shmool) [inline src/environments/environment.ts]: firebase.config was removed. Instead the credentials are in the environment files. We can (and should) have 2 different firebase projects for dev and prod, then we can put different credentials in each.

File: src/app/rappid-components/rappid-toolbar/rappid-toolbar.component.ts (hunk 1 above)
Comment (shmool) [inline src/app/rappid-components/rappid-toolbar/rappid-toolbar.component.ts]: it's static configuration and pretty short and straight forward. When it becomes more complex we'll do that.

**Pull request merged**